src/runway/commands/modules_command.py

```diff
@@ -520,26 +522,45 @@ def _process_modules(self, deployment, context):
                 self._deploy_module(module, deployment, context)
 
     def _deploy_module(self, module, deployment, context):
-        module_opts = {}
-        if deployment.get('environments'):
-            module_opts['environments'] = deployment['environments'].copy()  # noqa
-        if deployment.get('module_options'):
-            module_opts['options'] = deployment['module_options'].copy()  # noqa
-        if isinstance(module, six.string_types):
-            module = {'path': module}
-        if path_is_current_dir(module['path']):
+        """Execute module deployment.
+
+        1. Resolves variables in :class:`runway.config.DeploymentDefinition`
+           and :class:`runway.config.ModuleDefinition`.
+        2. Constructs a ``Dict`` of options to be passed to the ``module_class``.
+        3. Determine the class to use to execute the
+           :class:`runway.config.ModuleDefinition`, ``cd`` to the module
+           directory, and instanteate the class.
+        4. Find and execute the command method of the instanteated class.
+
+        Args:
+            module (:class:`runway.config.ModuleDefinition`): The module
+                to be deployed.
+            deployment (:class:`runway.config.DeploymentDefinition`): The
+                deployment the module belongs to. Used to get options,
+                environments, and env_vars from the deployment level.
+            context: (:class:`runway.context.Context`): Current context instance.
+
+        """
+        deployment.resolve(context, self.runway_vars)
+        module.resolve(context, self.runway_vars)
+        module_opts = {
+            'environments': deployment.environments.copy(),
+            'options': deployment.module_options.copy()
+        }
+        if path_is_current_dir(module.path):
             module_root = self.env_root
         else:
-            module_root = os.path.join(self.env_root, module['path'])
-        module_opts = merge_dicts(module_opts, module.__dict__)
+            module_root = os.path.join(self.env_root, module.path)
+        module_opts = merge_dicts(module_opts, module.data)
         module_opts = load_module_opts_from_file(module_root, module_opts)
 
         LOGGER.info("")
         LOGGER.info("---- Processing module '%s' for '%s' in %s --------------",
-                    module['path'],
+                    module.path,
                     context.env_name,
                     context.env_region)
         LOGGER.info("Module options: %s", module_opts)
+        module_opts['_using_vars'] = bool(self.runway_vars)
```

Comment:
@ITProKyle as part of our config discussion: I'd like to refactor so this isn't needed (i.e. the stacker module just uses the "parameter" values provided to it)